### PR TITLE
Add LR scheduler and Fix Granite Checkpoint Loading

### PR DESCRIPTION
**Pull request by fabianlim**
This PR adds the LR scheduler configuration and also fixes the granite loading issue for multinode settings.
- this PR subsumes the changes in @aldo-pareja 's `lr_scheduler` branch, so a seperate merge of that would not be necessary
- the strategy now for converting the granite checkpoint, is that each node will elect one local rank to do the convert.
- the conversion will be one on the `args.output_dir`. There is no assumption it is the same for all nodes or not.  But assuming they might be the same, the `tmpdir` is shared by a `.{group_rank}` postfix. See [group rank](https://pytorch.org/docs/stable/elastic/run.html#environment-variables)
- tested on multi node

### CODE REVIEW

**Review by Maxusmusti — APPROVED** [review]
LGTM, tested on single-node as well and worked. @aldo-pareja have you tested on multi-node?
